addons/hr_expense/models/product_template.py

```diff
@@ -30,6 +31,12 @@ def _auto_init(self):
             )
         return super()._auto_init()
 
-    @api.depends('type')
+    @api.depends('type', 'purchase_ok')
     def _compute_can_be_expensed(self):
-        self.filtered(lambda p: p.type not in ['consu', 'service']).update({'can_be_expensed': False})
+        self.filtered(lambda p: p.type not in ['consu', 'service'] or not p.purchase_ok).update({'can_be_expensed': False})
+
+    @api.depends('can_be_expensed')
+    def _compute_purchase_ok(self):
+        for record in self:
+            if record.can_be_expensed:
+                record.purchase_ok = True
```

Comment:
`record.purchase_ok = record.can_be_expensed`

Reply:
but the default for a boolean is False if we don't specify anything, so `record.purchase_ok = record.can_be_expensed` is equivalent to what you have now, no ?

Reply:
yeah but what if `can_be_expensed` is set to False? This will set `purchase_ok` to False as well , and as far as I understand, this is not the needed behavior.